eMPRess_GUI_version1.py

```diff
@@ -0,0 +1,354 @@
+
+    def DTL_cost(self):
+        """Sets DTL costs by clicking on the matplotlib graph or by entering manually."""
+        # creates a frame for setting DTL costs
+        costs_frame = tk.Frame(self.output_frame)
+        costs_frame.grid(row=1, column=0, sticky="nsew")
+        costs_frame.grid_columnconfigure(0, weight=1)
+        costs_frame.grid_columnconfigure(1, weight=3)
+        costs_frame.grid_rowconfigure(0, weight=1)
+        costs_frame.grid_rowconfigure(1, weight=1)
+        costs_frame.grid_rowconfigure(2, weight=1)
+        costs_frame.grid_propagate(False)
+
+        dup_label = tk.Label(costs_frame, text="Duplication:")
+        trans_label = tk.Label(costs_frame, text="Transfer:")
+        loss_label = tk.Label(costs_frame, text="Loss:")
+        dup_label.grid(row=0, column=0, sticky="w")
+        trans_label.grid(row=1, column=0, sticky="w")
+        loss_label.grid(row=2, column=0, sticky="w")
+        self.dup_cost = tk.StringVar()
+        self.trans_cost = tk.StringVar()
+        self.loss_cost = tk.StringVar()
+        dup_entry = tk.Entry(costs_frame, width=3, textvariable=self.dup_cost)
+        trans_entry = tk.Entry(costs_frame, width=3, textvariable=self.trans_cost)
+        loss_entry = tk.Entry(costs_frame, width=3, textvariable=self.loss_cost)
+        dup_entry.bind('<Return>', self.check_dup_input)
+        trans_entry.bind('<Return>', self.check_trans_input)
+        loss_entry.bind('<Return>', self.check_loss_input)
+        #dup_entry.bind('<FocusOut>', self.check_dup_input)  # (bug) this will make the invalid input warning appear twice
+        #trans_entry.bind('<FocusOut>', self.check_trans_input)  # same as above 
+        #loss_entry.bind('<FocusOut>', self.check_loss_input)  # same as above 
+        dup_entry.grid(row=0, column=1, sticky="w")
+        trans_entry.grid(row=1, column=1, sticky="w")
+        loss_entry.grid(row=2, column=1, sticky="w")
+
+    def plot_cost_landscape(self):
+        """Plots the cost landscape using matplotlib and embeds the graph in a tkinter window."""    
+        # creates a new tkinter window 
+        plt_window = tk.Toplevel(self.master)
+        plt_window.geometry("550x550")
+        plt_window.title("Matplotlib Graph DEMO")
+        # creates a new frame
+        plt_frame = tk.Frame(plt_window)
+        plt_frame.pack(fill=tk.BOTH, expand=1)
+        plt_frame.pack_propagate(False)
+        # use empress wrapper
+        recon_input = empress.read_input(self.file_path)
+        cost_regions = empress.compute_cost_regions(recon_input, 0.5, 10, 0.5, 10)  
+        #cost_regions.draw_to_file('./examples/cost_poly.png')  # draw and save to a file
+        fig = cost_regions.draw()  # draw to figure (creates matplotlib figure)
+        canvas = FigureCanvasTkAgg(fig, plt_frame)
+        canvas.draw()
+        canvas.get_tk_widget().pack(side=tk.TOP, fill=tk.BOTH, expand=True)
+        toolbar = NavigationToolbar2Tk(canvas, plt_frame)
```

Comment:
Does this toolbar allow users to zoom in/out? It might be a good idea to add a small comment what the toolbar enables the user to do.

Reply:
Yes, it does. I will add the comment. The toolbar allows the user to zoom in/out, drag the graph and save the graph.